src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -32,122 +45,200 @@ class ScatterPlot extends Visualization {
         let height    = this.dom.getAttributeNS(null, "height");
         const divElem = this.createDivElem(width, height);
 
-        let margin = {top: 20, right: 20, bottom: 20, left: 20};
-        width      = width - margin.left - margin.right;
-        height     = height - margin.top - margin.bottom;
+        let parsedData = JSON.parse(data);
+        let axis       = parsedData.axis || {x: {scale: "linear" }, y: {scale: "linear" }};
+        let focus      = parsedData.focus;
+        let points     = parsedData.points || {labels: "invisible"};
+        let dataPoints = parsedData.data || {};
+
+        let margin     = this.getMargins(axis);
+        let box_width  = width - margin.left - margin.right;
+        let box_height = height - margin.top - margin.bottom;
 
+        // FIXME : SVG eagerly gets all pointer events from top of it, even if
+        //         node overlaps it. Should be debugged with (#801).
         let svg = d3.select(divElem)
             .append("svg")
-            .attr("width", width + margin.left + margin.right)
-            .attr("height", height + margin.top + margin.bottom)
+            .attr("width", width)
+            .attr("height", height)
             .append("g")
             .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
 
-        // FIXME : SVG eagerly gets all pointer events from top of it, even if
-        //         node overlaps it. Should be debugged with (#797).
-
-        let parsedData  = JSON.parse(data);
-        let dataSource  = parsedData.source || "";
-        let colorDomain = []
-        let colorRange  = []
-        if (parsedData.colors !== undefined) {
-            parsedData.colors.forEach(d => {
-                colorDomain.push(d.domain);
-                colorRange.push("#" + d.color);
-            });
-        }
+        let {xMin, xMax, yMin, yMax, padding_x, padding_y} = this.getExtremesAndDeltas(dataPoints);
 
-        d3.csv(dataSource, this.presentDataOnScatterplot(width, svg, height, colorDomain, colorRange));
+        let {x, xAxis, y} = this.createAxes(axis, xMin, padding_x, xMax, box_width, svg, box_height, yMin, padding_y, yMax);
+
+        this.createLabels(axis, svg, box_width, margin, box_height);
+
+        let scatter = this.createSymbols(svg, box_width, box_height, points, dataPoints, x, y);
```

Comment:
So, scatter is a structure with symbols? If yes, I would make `scatter = this.createScatter` or `symbols = this.createSymbols` (I prefer the former).